common/config/environment.go

```diff
@@ -166,6 +171,24 @@ func (e *CommonEnvironment) GetIntWithDefault(config *sdkconfig.Config, paramNam
 	return defaultValue
 }
 
+func (e *CommonEnvironment) CommandProvider(namer namer.Namer, name string) (*command.Provider, error) {
```

Comment:
You don't need to take any parameter in, `Provider` are not currently named with namer for simplicity: https://github.com/DataDog/test-infra-definitions/blob/bf98888bd09fe6e3e760f49f6c0fb34e146ab36f/aws/environment.go#L73-L86

Also I don't think we should return an error on that, it will allow to do `Pulumi.Provider(e.CommandProvider())`

You can `panic` if you get an error there.